backend/tests/e2e/test_e2e_nlweb_mcp.py

```
@@ -0,0 +1,166 @@
+    mock_capability_service.register_capability(test_capability_metadata)
+    app.dependency_overrides[get_a2a_capability_service] = lambda: mock_capability_service
+
+    #   b. Mock A2AHandlerService for the actual execution logic of 'test_processor'
+    mock_handler_service = MagicMock(spec=A2AHandlerService)
+    input_text_for_mcp_test = "hello mcp a2a"
+    expected_handler_output = {
+        "processed_text": f"mocked_for_mcp: {input_text_for_mcp_test}",
+        "status_detail": "successfully processed by mock A2A handler via MCP call"
+    }
+
+    async def mock_handle_a2a_request_for_mcp(capability_name: str, data: BaseModel) -> Dict[str, Any]:
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)